src/ode_premade_problems.jl

```diff
@@ -375,3 +375,574 @@ end
+    du[1,1,2] = α*(2u[i+1,j,2] + 2u[i,j+1,2] - 4u[i,j,2]) +
+    A*u[i,j,1] - u[i,j,1]^2*u[i,j,2]
+
+    i = 1; j = N
+    du[1,N,1] = α*(2u[i+1,j,1] + 2u[i,j-1,1] - 4u[i,j,1]) +
+    B + u[i,j,1]^2*u[i,j,2] - (A + 1)*u[i,j,1]
+    du[1,N,2] = α*(2u[i+1,j,2] + 2u[i,j-1,2] - 4u[i,j,2]) +
+    A*u[i,j,1] - u[i,j,1]^2*u[i,j,2]
+
+    i = N; j = 1
+    du[N,1,1] = α*(2u[i-1,j,1] + 2u[i,j+1,1] - 4u[i,j,1]) +
+    B + u[i,j,1]^2*u[i,j,2] - (A + 1)*u[i,j,1]
+    du[N,1,2] = α*(2u[i-1,j,2] + 2u[i,j+1,2] - 4u[i,j,2]) +
+    A*u[i,j,1] - u[i,j,1]^2*u[i,j,2]
+
+    i = N; j = N
+    du[end,end,1] = α*(2u[i-1,j,1] + 2u[i,j-1,1] - 4u[i,j,1]) +
+    B + u[i,j,1]^2*u[i,j,2] - (A + 1)*u[i,j,1]
+    du[end,end,2] = α*(2u[i-1,j,2] + 2u[i,j-1,2] - 4u[i,j,2]) +
+    A*u[i,j,1] - u[i,j,1]^2*u[i,j,2]
+  end
+end
+function init_brusselator_2d(xyd)
+  M = length(xyd)
+  u = zeros(M, M, 2)
+  for I in CartesianRange((M, M))
+    u[I,1] = (2 + 0.25xyd[I[2]])
+    u[I,2] = (1 + 0.8xyd[I[1]])
+  end
+  u
+end
+const N_brusselator_2d = 128
+prob_ode_brusselator_2d = ODEProblem(brusselator_2d_loop,
+                                     init_brusselator_2d(linspace(0,1,N_brusselator_2d)),
+                                     (0.,1),
+                                     (3.4, 1., 0.002, 1/(N_brusselator_2d-1),
+                                     N_brusselator_2d))
+
+const N_brusselator_1d = 40
+const D_brusselator_u = DerivativeOperator{Float64}(2,2,1/(N_brusselator_1d-1),
+                                                    N_brusselator_1d,
+                                                    :Dirichlet,:Dirichlet;
+                                                    BC=(1.,1.))
+const D_brusselator_v = DerivativeOperator{Float64}(2,2,1/(N_brusselator_1d-1),
+                                                    N_brusselator_1d,
+                                                    :Dirichlet,:Dirichlet;
+                                                    BC=(3.,3.))
+function brusselator_1d(du, u_, p, t)
+    A, B, α, buffer = p
+    u = @view(u_[:, 1])
+    v = @view(u_[:, 2])
+    A_mul_B!(buffer, D_brusselator_u, u)
+    Du = buffer
+    @. du[:, 1] = A + u^2*v - (B+1)*u + α*Du
+
+    A_mul_B!(buffer, D_brusselator_v, v)
+    Dv = buffer
+    @. du[:, 2] = B*u - u^2*v + α*Dv
+    nothing
+end
+function init_brusselator_1d(N)
+  u = zeros(N, 2)
+  x = linspace(0, 1, N)
+  for i in 1:N
+    u[i, 1] = 1 + sin(2pi*x[i])
+    u[i, 2] = 3.
+  end
+  u
+end
+prob_ode_brusselator_1d = ODEProblem(brusselator_1d,
+                                    init_brusselator_1d(N_brusselator_1d),
+                                    (0.,10.),
+                                    (1., 3., 1/50, zeros(N_brusselator_1d)))
+
+orego = @ode_def Orego begin
+  dy1 = p1*(y2+y1*(1-p2*y1-y2))
+  dy2 = (y3-(1+y1)*y2)/p1
+  dy3 = p3*(y1-y3)
+end p1 p2 p3
+prob_ode_orego = ODEProblem(orego,[1.0,2.0,3.0],(0.0,30.0),[77.27,8.375e-6,0.161])
+
+hires = @ode_def Hires begin
```

Comment:
we should probably set these up to use a tuple of parameters so that way they are good parameter estimation benchmarks too.